gef-ui/server/controllers/bank-review-date/get-bank-review-date.ts

```diff
@@ -0,0 +1,87 @@
+import { Response } from 'express';
+import { getDate, getMonth, getYear, parseISO } from 'date-fns';
+import { CustomExpressRequest, isFacilityEndDateEnabledOnGefVersion, parseDealVersion } from '@ukef/dtfs2-common';
+import * as api from '../../services/api';
+import { BankReviewDateViewModel } from '../../types/view-models/bank-review-date-view-model';
+import { asLoggedInUserSession } from '../../utils/express-session';
+import { Facility } from '../../types/facility';
+
+type GetBankReviewDateRequest = CustomExpressRequest<{ params: { dealId: string; facilityId: string }; query: { status: string | undefined } }>;
+
+type HandleGetBankReviewDateParams = {
+  req: GetBankReviewDateRequest;
+  res: Response;
+  previousPage: string;
+};
+
+/**
+ * @param facility - the facility
+ * @param previousPage - the previous page url
+ * @param status - the query parameter status
+ * @returns view model for the bank review date template
+ */
+const getBankReviewDateViewModel = (facility: Facility, previousPage: string, status?: string): BankReviewDateViewModel => {
+  const bankReviewDateViewModel: BankReviewDateViewModel = {
+    dealId: facility.dealId,
+    facilityId: facility._id,
+    previousPage,
+    status,
+  };
+
+  if (facility.bankReviewDate) {
+    const bankReviewDate = parseISO(facility.bankReviewDate);
+    bankReviewDateViewModel.bankReviewDate = {
+      day: String(getDate(bankReviewDate)),
+      month: String(getMonth(bankReviewDate) + 1),
+      year: String(getYear(bankReviewDate)),
+    };
+  }
+
+  return bankReviewDateViewModel;
+};
+
+/**
+ * Handle get bank review date requests
+ */
+const getBankReviewDate = async ({ req, res, previousPage }: HandleGetBankReviewDateParams) => {
+  const {
+    params: { dealId, facilityId },
+    query: { status },
+  } = req;
+  const { userToken } = asLoggedInUserSession(req.session);
+
+  try {
+    const { details: facility } = await api.getFacility({ facilityId, userToken });
+    const deal = await api.getApplication({ dealId, userToken });
+
+    const dealVersion = parseDealVersion(deal.version);
+    const shouldRedirectFromPage = !isFacilityEndDateEnabledOnGefVersion(parseDealVersion(dealVersion)) || facility.isUsingFacilityEndDate !== false;
```

Comment:
There's a duplicate `parseDealVersion` here. Also looks like the "not false" check could just check if its true?

```suggestion
  const shouldRedirectFromPage = !isFacilityEndDateEnabledOnGefVersion(dealVersion) || facility.isUsingFacilityEndDate;
```